troubleshooting/incrementally_configuring_images.md

```diff
@@ -0,0 +1,344 @@
+
+   ```bash
+   cray cfs v3 sessions create --name $SESSION_NAME --configuration-name $CFS_CONFIGURATION_NAME --target-definition image --target-group Compute $BASE_IMAGE_ID --configuration-limit $CONFIGURATION_LIMIT
+   ```
+
+   If this is the last layer of the configuration, it is optionally possible to specify the name of the resulting image by adding `--target-image-map`.
+
+   ```bash
+   cray cfs v3 sessions create --name $SESSION_NAME --configuration-name $CFS_CONFIGURATION_NAME --target-definition image --target-group Compute $BASE_IMAGE_ID --configuration-limit $CONFIGURATION_LIMIT --target-image-map $BASE_IMAGE_ID <desired image name>
+   ```
+
+1. (`ncn-mw#`) Monitor CFS and retrieve the partial image id.
+
+    Monitor the CFS session with the following command until the status is complete.  If `suceeded` is true, move on to the next step, otherwise debug the failure and re-run the session as necessary.
+
+   ```bash
+   cray cfs sessions describe $SESSION_NAME --format json | jq .status.session
+   ```
+
+   Example output:
+
+   ```json
+   {
+     "completionTime": "2023-11-01T19:55:10",
+     "job": "cfs-712dee37-2b80-498c-867c-42753716cad6",
+     "startTime": "2023-11-01T19:53:09",
+     "status": "complete",
+     "succeeded": "true"
+   }
+   ```
+
+1. When the session is complete, get the resulting IMS image id.
+
+   ```bash
+   cray cfs v3 sessions describe $SESSION_NAME --format json | jq .status.artifacts
+   ```
+
+   Example output:
+
+   ```json
+   [
+     {
+       "image_id": "<IMS IMAGE ID>",
+       "result_id": "<RESULTANT IMS IMAGE ID>",
+       "type": "ims_customized_image"
+     }
+   ]
+   ```
+
+1. Prepare for the next configuration layer.
+
+   If this was not the last layer in the configuration, set `BASE_IMAGE_ID` equal to the `result_id` from the last CFS session, and cleanup the CFS session.
+   Then return to the top of this section.
+   If this was the last layer in the configuration, move on to the next section.
+
+   ```bash
+   export BASE_IMAGE_ID=<RESULTANT IMS IMAGE ID>
+   cray cfs v3 sessions delete $SESSION_NAME
+   ```
+
+## Restarting SAT and IUF
+
+1. Create a copy of the original SAT `bootprep` file.
+
+   Instructions for this procedure can be found in the [Stage 0](../upgrade/Stage_0_Prerequisites.md#option-2-upgrade-of-csm-on-system-with-additional-products) upgrade docs.
+
+1. Edit the session templates in the SAT `bootprep` file.
+
+   For any BOS templates in the `session_templates` section that will now be booting with the image you have just created, replace `image_ref` with `image`:
+
+   Before:
+
+   ```yaml
+   session_templates:
+   - name: example-session-template
+     image:
+       image_ref: example-image
+   ```
+
+   After:
+
+   ```yaml
+   session_templates:
+   - name: example-session-template
+     image:
+       ims:
+         id: <image id from the final CFS run>
+   ```
+
+1. Remove configuration creation from the SAT `bootprep` file.
```

Comment:
If they restart from the `prepare-images` stage, they shouldn't need to do this since that stage of the IUF runs `sat bootprep` with `--limit images --limit session_templates` and thus skips creating CFS configurations.